### PR TITLE
fix: requestlayout improvements

### PR DESCRIPTION
**Pull request by farfromrefug**
This is a first pass at improving `requestLayout`.  The biggest improvements are seen on iOS.
* during `applyAllNativeSetters` or `applyPendingNativeSetters` prevent `requestLayout` calls on every property change. Instead call `requestLayout` if needed at the end. This is a big change on iOS as most properties are marked as affecting layout. 
* on top of that dont even call `requestLayout` at all if within `onLoaded`. it is not necessary as the parent with layout itself and thus layout all its children.

With that change, the colors fixes and the android background color change simplification, i see a 5-10%  shorter loading time on android. Tested it with different apps of different complexity.

### CODE REVIEW

**Comment by NathanaelA** [comment]
This looks awesome.  Thanks -- did you run it against the app testing harness?

**Comment by farfromrefug** [comment]
Which app  are you referring to?

**Comment by NathanWalker** [comment]
@farfromrefug `npm start` > type `automated.ios` (and you can also run `automated.android`) to kick them off locally.

**Comment by farfromrefug** [comment]
@NathanWalker thanks was not sure @NathanaelA was refering to another one.

File: packages/core/ui/core/view-base/index.ts [hunk 1]
```diff
@@ -619,8 +623,24 @@ export abstract class ViewBase extends Observable implements ViewBaseDefinition
 		}
 	}
 
+
+	_requetLayoutNeeded = false;
```
Comment (NathanWalker) [inline packages/core/ui/core/view-base/index.ts:627]: @farfromrefug Could you just fix the spelling here:

```suggestion
	_requestLayoutNeeded = false;
```

File: packages/core/ui/core/view/index.ios.ts [hunk 2]
```diff
@@ -54,9 +54,21 @@ export class View extends ViewCommon implements ViewDefinition {
...
+		if ( this.isLayoutRequired) {
+			this._requetLayoutNeeded = false;
+			this.requestLayout();
+		}
+	}
+
 	public requestLayout(): void {
-		super.requestLayout();
+		if (this._suspendRequestLayout) {
+			this._requetLayoutNeeded = true;
```
Comment (NathanWalker) [inline packages/core/ui/core/view/index.ios.ts:66]:
```suggestion
			this._requestLayoutNeeded = true;
```

File: packages/core/ui/core/view/index.android.ts [hunk 3]
```diff
@@ -525,6 +525,11 @@ export class View extends ViewCommon {
 
 	@profile
 	public requestLayout(): void {
+		if (this._suspendRequestLayout) {
+			this._requetLayoutNeeded = true;
```
Comment (NathanWalker) [inline packages/core/ui/core/view/index.android.ts:529]:
```suggestion
			this._requestLayoutNeeded = true;
```

File: packages/core/ui/core/view/index.android.ts [hunk 4]
```diff
@@ -525,6 +525,11 @@ export class View extends ViewCommon {
...
+			return;
+		}
+		this._requetLayoutNeeded = false;
```
Comment (NathanWalker) [inline packages/core/ui/core/view/index.android.ts:532]:
```suggestion
		this._requestLayoutNeeded = false;
```

File: packages/core/ui/core/view/index.ios.ts [hunk 5]
```diff
@@ -54,9 +54,21 @@ export class View extends ViewCommon implements ViewDefinition {
 		return (this._privateFlags & PFLAG_FORCE_LAYOUT) === PFLAG_FORCE_LAYOUT;
 	}
 
+	requestlayoutIfNeeded() {
+		if ( this.isLayoutRequired) {
+			this._requetLayoutNeeded = false;
```
Comment (NathanWalker) [inline packages/core/ui/core/view/index.ios.ts:59]:
```suggestion
			this._requestLayoutNeeded = false;
```

File: packages/core/ui/core/view/index.ios.ts [hunk 6]
```diff
@@ -54,9 +54,21 @@ export class View extends ViewCommon implements ViewDefinition {
...
+		}
+	}
+
 	public requestLayout(): void {
-		super.requestLayout();
+		if (this._suspendRequestLayout) {
+			this._requetLayoutNeeded = true;
+			return;
+		}
+		this._requetLayoutNeeded = false;
```
Comment (NathanWalker) [inline packages/core/ui/core/view/index.ios.ts:69]:
```suggestion
		this._requestLayoutNeeded = false;
```

File: packages/core/ui/core/view-base/index.ts [hunk 7]
```diff
@@ -619,8 +623,24 @@ export abstract class ViewBase extends Observable implements ViewBaseDefinition
...
+
+	get isLayoutRequestNeeded() {
+		return this._requetLayoutNeeded;
```
Comment (NathanWalker) [inline packages/core/ui/core/view-base/index.ts:630]:
```suggestion
		return this._requestLayoutNeeded;
```

File: packages/core/ui/core/view/index.ios.ts [hunk 8]
```diff
@@ -54,9 +54,21 @@ export class View extends ViewCommon implements ViewDefinition {
 		return (this._privateFlags & PFLAG_FORCE_LAYOUT) === PFLAG_FORCE_LAYOUT;
 	}
 
+	requestlayoutIfNeeded() {
```
Comment (NathanWalker) [inline packages/core/ui/core/view/index.ios.ts:57]: Maintain same camel case throughout:
```suggestion
	requestLayoutIfNeeded() {
```

**Review by NathanWalker — CHANGES_REQUESTED** [review]

**Comment by farfromrefug** [comment]
@rigor789 should be all good

**Review by rigor789 — APPROVED** [review]
LGTM

**Comment by NathanWalker** [comment]
@farfromrefug I'm curious if your fork may have a different setting than most? I've noticed that on all your PR's we can not update the PR with base automatically on github however for all other pr's we are able to. Is there a setting on your fork that may be preventing that? We could help get quite a few of your pr's merged if we can update them with latest master automatically here.

**Comment by farfromrefug** [comment]
@NathanWalker you mean on my PR s you font have the right to update? I always create PRs from github website but I have to say I almost never ensure the checkbox for collaboration is checked or not. though it should be by default

**Pull request merged**